src/aave-v3/SupplyVault.sol

```diff
@@ -226,40 +203,59 @@ contract SupplyVault is ISupplyVault, SupplyVaultBase {
             (rewardTokens, claimedAmounts) = morpho.claimRewards(poolTokens, false);
         }
 
-        uint256 supply = totalSupply();
```

Comment:
couldn't this be kept in memory?

Reply:
Yes I think we could but it requires adding the total supply to the arguments of the `_getUnaccruedRewardIndex` function

Reply:
I mean it's still read in the loop so without talking about passing it down as a parameter, it could still be worth it to keep it in the stack at this level

Reply:
Oh, I thought this was related to moving it to a parameter. Yes you're right.